src/nvim/os/env.c

```diff
@@ -105,6 +105,10 @@ bool os_env_exists(const char *name)
   return (r == 0 || r == UV_ENOBUFS);
 }
 
+/// Sets an environment variable.
+///
+/// @warning Existing pointers to the result of os_getenv("foo") are
+///          INVALID after os_setenv("foo", …).
 int os_setenv(const char *name, const char *value, int overwrite)
```

Comment:
Future improvement: instead of returning `int`, we could return a pointer to the new `os_getenv()` result. So the caller pattern could be like:

    term = os_setenv("TERM", term, ...)

However, after inspecting of all existing calls to `os_setenv` and `vim_setenv`, it looks like `tui.c` was the only case where a `os_getenv` result was passed to `os_setenv`.